src/main/java/org/mule/extension/http/internal/request/HttpPollingSource.java

```diff
@@ -186,8 +211,48 @@ public void poll(PollContext<String, HttpResponseAttributes> pollContext) {
   }
 
   @Override
-  public void onRejectedItem(Result<String, HttpResponseAttributes> result, SourceCallbackContext sourceCallbackContext) {
-    LOGGER.debug("Item rejected by HTTP Polling Source in flow '{}' with result: '{}", location.getRootContainerName(),
+  public void onRejectedItem(Result<String, HttpResponseAttributes> result,
+                             SourceCallbackContext sourceCallbackContext) {
+    LOGGER.debug("Item rejected by HTTP Polling Source in flow '{}', result: '{}'", location.getRootContainerName(),
                  result.getOutput());
   }
+
+  private BindingContext buildContext(TypedValue<String> payload, HttpResponseAttributes attributes,
+                                      java.util.Optional<TypedValue<CursorStreamProvider>> item) {
+    BindingContext.Builder builder = BindingContext.builder().addBinding("payload", payload);
+    builder.addBinding("attributes", TypedValue.of(attributes));
+    item.ifPresent(it -> builder.addBinding("item", it));
+    return builder.build();
+  }
+
+  private static TypedValue<String> toTypedValue(String value, MediaType mediaType, Charset encoding) {
+    if (mediaType.equals(TEXT.withCharset(mediaType.getCharset().orElse(null)))) {
+      return TypedValue.of(value);
+    } else {
+      return new TypedValue<>(value, DataType.builder().mediaType(mediaType).charset(encoding).build());
+    }
+  }
+
+  private Result<String, HttpResponseAttributes> toResult(String item, MediaType mediaType, HttpResponseAttributes attributes) {
+    return Result.<String, HttpResponseAttributes>builder().attributes(attributes).output(item).mediaType(mediaType).build();
+  }
+
+  private List<Result<String, HttpResponseAttributes>> getItems(InputStream fullResponse, MediaType mediaType,
+                                                                HttpResponseAttributes attributes) {
+    java.util.Optional<String> itemsExpression = expressions.getSplitExpression();
+    Charset charset = mediaType.getCharset().orElse(defaultCharset());
+    String response = IOUtils.toString(fullResponse, charset);
+    LOGGER.debug("Received response at {}: {} and headers {}", location.getRootContainerName(), response,
+                 attributes.getHeaders());
+    if (!itemsExpression.isPresent()) {
+      return singletonList(toResult(response, mediaType, attributes));
+    }
+    TypedValue<String> typedValue = toTypedValue(response, mediaType, charset);
+    TypedValue<?> result = expressionLanguage.evaluate(itemsExpression.get(), buildContext(typedValue, attributes, null));
```

Comment:
The last parameter of buildContext is an Optional and here you're passing a null. It'll lead to a NPE